saleor/discount/utils.py

```diff
@@ -86,3 +88,11 @@ def get_products_voucher_discount(voucher, prices):
         voucher.get_discount_amount_for(price) for price in prices)
     total_amount = sum(discounts, ZERO_MONEY)
     return total_amount
+
+
+def generate_voucher_code():
+    """Generate new unique voucher code."""
+    while True:
```

Comment:
😱 I just realized that we've had this ugly `while True` for so long. Can we get rid of this line? Chances of collision in UUID are so low that we could ignore it (we do for generating order tokens for example).

Reply:
In addition, if an impossible was happen, we wouldn't store the duplicate in database (due to `unique=True`). 
I've removed the loop.